### ISSUE TITLE
Code review

### ISSUE THREAD

**Issue by frank613** [e-c-k-e-r/vall-e#9]
Hi Ecker,

I am reading your new version of code. In nar_mdd_v2.py, line 363
`

	if score_masked_only:
		scores = [ torch.where( masked, scores.t(), smallest_score ) for masked, scores in zip( is_masked, sampled.scores ) ]

	else:
		scores = [ scores.t() for scores in sampled.scores ]`

In the first case, the frames are assigned with the "smallest_score" (0) if they are  _**not**_ masked. However, in the next iteration:
`masked_indices = [ score.topk( clamp( int( mask_p * seq_len ), 1, seq_len - step), dim=0, largest=False ).indices for score, seq_len in zip(scores, len_list) ]`

it selects those "smallest topK" frames for masking and further for updating. It does not mean "force keeping unmasked tokens". Instead, I think it might lead to predicting already good enough frames "back and forth".  Maybe I am wrong at some point?

Thanks

**Comment by e-c-k-e-r** [comment]
Oh. I'm fairly confident this is an error/relic from when I was flipping over to `largest=False` top-k scoring/masking. From what I could remember from my last experiments a few weeks ago (or a month ago, I can't remember as I've had other endeavors require my attention), this should explain why `score_masked_only=True` had some problems in the output for the `nvidia/audio-codec-44khz` model. Thankfully I have it default to `score_masked_only=False`, including in the web UI.

I'll push a fix when I get a chance.
